gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspConfigurations.kt

```diff
@@ -0,0 +1,136 @@
+package com.google.devtools.ksp.gradle
+
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.jetbrains.kotlin.gradle.dsl.*
+import org.jetbrains.kotlin.gradle.plugin.*
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinJvmAndroidCompilation
+
+/**
+ * Creates and retrieves ksp-related configurations.
+ */
+class KspConfigurations(private val project: Project) {
+    companion object {
+        private const val PREFIX = "ksp"
+    }
+
+    // "ksp" configuration, applied to the "main" source set of all targets.
+    private val rootMainConfiguration = project.configurations.create(PREFIX)
+
+    // Store all ksp configurations for quick retrieval.
+    private val kotlinConfigurations = mutableMapOf<KotlinSourceSet, Configuration>()
+    private val androidConfigurations = mutableMapOf<String, Configuration>()
+
+    @OptIn(ExperimentalStdlibApi::class)
+    private fun <T : Any> saveConfiguration(
+        owner: T,
+        parent: Configuration?,
+        name: String,
+        cache: MutableMap<T, Configuration>
+    ): Configuration {
+        val configName = PREFIX + name.replaceFirstChar { it.uppercase() }
+        val config = if (configName == parent?.name) {
+            // trying to add config with same parent name. Can happen, for example
+            // with ksp<Target> when <Target> is "". Just use parent.
+            parent
+        } else {
+            // maybeCreate to be future-proof, but we should never have a duplicate with current logic
+            project.configurations.maybeCreate(configName).apply {
```

Comment:
Can you please add the following properties to the configuration:
- `description` e.g `KSP dependencies for jvmTest source set`
- set `setVisible(false)`
- set `canBeResolved = false` (as we should never resolve these directly)
- set `isCanBeConsumed = false` (to make sure this configuration is not used for publishing)